### PR TITLE
Feature/ws secutiry for admin agent communications epmdj 10639

### PR DESCRIPTION
**Pull request by epamrsa**

### CODE REVIEW

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt [hunk 1]
```diff
@@ -141,9 +140,9 @@ object Agent {
         val adminAddress = get("adminAddress")
         checkNotNull(adminAddress) { "Please set 'adminAddress' as agent parameters e.g. -agentpath:/path/to/agent=adminAddress={hostname:port}" }
         try {
-            URL("ws://$adminAddress")
+            URL(adminAddress)
         } catch (parseException: RuntimeException) {
-            fail("Please check 'adminAddress' parameter. It should be a valid address to the admin service without schema and any additional paths, e.g. localhost:8090")
+            fail("Please check 'adminAddress' parameter. It should be a valid address to the admin service any additional paths, e.g. ws://localhost:8090")
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/Agent.kt:145]: add "without" word back

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt [hunk 2]
```diff
@@ -96,6 +98,16 @@ data class AgentArguments(
             type = packagePrefixes.toType(),
             value = packagePrefixes,
             description = "Configure package prefixes for scanning and instrumentation",
+        ),
+        AgentArguments::sslTruststore.name to AgentParameter(
+            type = sslTruststore.toType(),
+            value = sslTruststore,
+            description = "Configure SSL truststore for admin connection (leave empty to thrust all)",
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/AgentArguments.kt:105]: 1. Change "thrust" to "trust" 😃 
2. Is it the _path_ to truststore, or is it it's name? (if its a _path_, specify it in the description. I.e. "Configure path to SSL truststore")

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/jvmti/event/VmInitEvent.kt [hunk 3]
```diff
@@ -55,8 +55,8 @@ fun vmInitEvent(env: CPointer<jvmtiEnvVar>?, jniEnv: CPointer<JNIEnvVar>?, threa
     globalCallbacks()
     updatePackagePrefixesConfiguration()
     loadJvmModule("test2code")
-    WsSocket().connect(adminAddress.toString())
     RequestHolder.init(isAsync = agentParameters.isAsyncApp)
+    WsClient.connect(adminAddress.toString(), true)
```
Comment (RomanDavlyatshin) [inline java-agent/src/nativeMain/kotlin/com/epam/drill/agent/jvmti/event/VmInitEvent.kt:59]: Will we ever pass `false`?

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: java-agent/src/nativeMain/kotlin/com/epam/drill/agent/jvmti/event/VmInitEvent.kt (hunk 3 above)
Comment (epamrsa) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/jvmti/event/VmInitEvent.kt:59]: This concerns connect() and reconnect() method, they use similar signature with async param with default false value for both methods. For connect() we are using async=true, but for reconnect we are leaving it default.

Comment (RomanDavlyatshin) [reply java-agent/src/nativeMain/kotlin/com/epam/drill/agent/jvmti/event/VmInitEvent.kt:59]: Added for expressiveness - connect is always in the new thread, reconnect in "current" thread

**Review by RomanDavlyatshin — APPROVED** [review]

**Pull request merged**